src/main/java/edu/utah/ece/async/sboldesigner/sbol/editor/dialog/VariantEditor.java

```diff
@@ -498,24 +529,32 @@ public void actionPerformed(ActionEvent e) {
 
 			if (e.getSource() == addButton) {
 				ComponentDefinitionBox root = new ComponentDefinitionBox();
-				SBOLDocument selection = new RegistryInputDialog(parent, root, Parts.forIdentified(variableCD),
-						SBOLUtils.Types.DNA, null, design).getInput();
+				RegistryInputDialog dialog = new RegistryInputDialog(parent, root, Parts.forIdentified(variableCD),
+						SBOLUtils.Types.DNA, null, design);
+				dialog.allowCollectionSelection();
+				SBOLDocument selection = dialog.getInput();
 
 				if (selection == null) {
 					return;
 				}
 
 				SBOLUtils.insertTopLevels(selection, design);
-				addVariant(root.cd);
+				if(selection.getComponentDefinitions().isEmpty() && !selection.getCollections().isEmpty())

```

Comment:
Also need an ability to add a variantDerivation.

Reply:
If getCollections is non-empty, then add the "root" collection as a variantCollection (need to loop through all Collections, make sure not a member of another collection). 

else if getCombinatorialDerivations is non-empty, then add all the root CombinatorialDerivations as variantDerivations.

else add the root componentDefinitions as variants.